namer/consul/src/test/scala/io/buoyant/namer/consul/SvcAddrTest.scala

```diff
@@ -150,4 +150,84 @@ class SvcAddrTest extends FunSuite with Matchers with Awaits {
     addr.lastUpdatedAt shouldBe 'defined
     addr.underlying.sample() should matchPattern { case Addr.Neg => }
   }
+
+  test("should use last known state on api timeout") {
+    // given
+    val requestCounter = new AtomicInteger()
+    val (initServiceUpdate, firstAddr) = service()
+    val (secondSvcUpdate, secondAddr) = service("9.9.9.9", 1024)
+    val ttl = 1.minute
+    @volatile var changes: Addr = Addr.Pending
+
+    val timer = new MockTimer
+
+    val api = apiStub { (_, _, _, _, _, _) =>
+      val count = requestCounter.incrementAndGet()
+      count match {
+        case 1 =>
+          val response = Indexed[Seq[ServiceNode]](Seq(initServiceUpdate), Some("1"))
+          Future.sleep(5.minutes)(timer).before(Future.value(response))
+        case 2 =>
+          val rspF = Future.sleep(11.minutes)(timer)
+            .before(Future.exception(new IndividualRequestTimeoutException(10.minutes)))
+          rspF
+        case 3 =>
+          Future.sleep(1.minute)(timer)
+            .before(Future.value(Indexed[Seq[ServiceNode]](Seq(secondSvcUpdate), Some("2"))))
+      }
+    }
+
+    // when
+    Time.withCurrentTimeFrozen { tc =>
+
+      val addr: InstrumentedVar[Addr] = SvcAddr(
+        api,
+        Stream.fill(10)(ttl),
+        "dc1",
+        SvcKey("svc", None),
+        None,
+        None,
+        None,
+        Map.empty,
+        Stats(NullStatsReceiver),
+        new PollState
+      )(timer)
+
+      addr.underlying.changes.respond {
+        changes = _
+      }
+
+      // then
+      tc.advance(5.minutes)
+      timer.tick()
+      changes match {
+        case Addr.Bound(addrs, _) => addrs.head shouldBe Address
+          .Inet(firstAddr, Addr.Metadata("endpoint_addr_weight" -> 1.0))
+        case _ => fail("received unexpected Addr on initial service discovery")
+      }
+
+      tc.advance(12.minutes)
+      timer.tick()
+      changes match {
+        case Addr.Bound(addrs, _) => addrs.head shouldBe Address
+          .Inet(firstAddr, Addr.Metadata("endpoint_addr_weight" -> 1.0))
+        case _ => fail("received unexpected Addr on timed out service discovery request")
+      }
+
+      // Advance timer to go past backoff sleep future in SvcAddr
+      tc.advance(5.minutes)
```

Comment:
Do we need both of this `tc.advance` call and the next one?  Should they be combined?